plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -624,6 +625,27 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .iterator();
     }
 
+    private static Schema getCurrentSnapshotSchema(Table icebergTable)
+    {
+        Optional<Long> tableSnapshotId = Optional.ofNullable(icebergTable.currentSnapshot()).map(Snapshot::snapshotId);
+        // Check whether the snapshot is latest or not because changing table definition (e.g. ADD COLUMN) doesn't generate a new snapshot
+        // If we use schema for 'tableSnapshotId' without this comparison, it returns old table definition which isn't different from the latest one
+        if (tableSnapshotId.isPresent() && !findLatestSnapshotId(icebergTable.snapshots()).equals(tableSnapshotId)) {
+            return schemaFor(icebergTable, tableSnapshotId.get());
+        }
+        return icebergTable.schema();
+    }
+
+    private static Optional<Long> findLatestSnapshotId(Iterable<Snapshot> snapshots)
+    {
+        try {
+            return Optional.of(Iterables.getLast(snapshots).snapshotId());
```

Comment:
Is `icebergTable.snapshots()` guaranteed to be ordered, or we need to sort this first?

(see https://github.com/trinodb/trino/pull/10258#discussion_r881655341 )